backend/server.js

```diff
@@ -538,6 +527,175 @@ app.post('/subscribeEmail', wrap(async (req, res) => {
 }));
 
 
+// finds the user with the login key that's been requested
+// if the user doesn't exist, return
+async function findMatchingUser(requestLoginKey) {
+  // Loop over the db
+  const users = await database.get('users');
+  if (!users) {
+    return null;
+  }
+
+  // Loop over all the users
+  for (const user of users) {
+    for (const loginKey of user.loginKeys) {
+      if (requestLoginKey === loginKey) {
+        return user;
+      }
+    }
+  }
+
+  return null;
+}
+
+// sends data to the database in the backend
+app.post('/sendUserData', wrap(async (req, res) => {
+  // Don't cache this endpoint.
+  res.setHeader('Cache-Control', 'no-cache, no-store');
+
+
+  // if there's no body in the request, well, we'll crash, so let's not
+  if (!req.body || !req.body.loginKey) {
+    res.send(JSON.stringify({
+      error: 'Error.',
+    }));
+    return;
+  }
+
+  // Checks checks checks
+  // Make sure the login key is valid
+  if (typeof req.body.loginKey !== 'string' || req.body.loginKey.length !== 100) {
+    macros.log('Invalid login key', req.body.loginKey);
+    res.send(JSON.stringify({
+      error: 'Error.',
+    }));
+    return;
+  }
+
+  const senderId = req.body.senderId;
+
+  // Make sure the sender id is valid
+  if (senderId && (typeof senderId !== 'string' || senderId.length !== 16 || !macros.isNumeric(senderId))) {
+    macros.log('Invalid senderId', req.body, senderId);
+    res.send(JSON.stringify({
+      error: 'Error.',
+    }));
+    return;
+  }
+
+
+  // If the client specified a specific senderId, lookup that specific user.
+  // if not, we have to loop over all the users's to find a matching loginKey
+  if (senderId) {
+    // first confirm the loginkey is legitimate
+    const user = await database.get(`/users/${senderId}`);
+    if (user && !user.loginKeys.includes(req.body.loginKey)) {
+      macros.log('loginKey not within the sender\'s entry');
+      res.send(JSON.stringify({
+        error: 'Error, invalid loginkey on senderId.',
+      }));
+      return;
+    }
+    await database.set(`/users/${senderId}`, req.body.info);
+    macros.log('sending done, result is ', await database.get(`/users/${senderId}`));
```

Comment:
remove this line when you're done with everything - the logging is find just don't want the DB call